benchmarks/314.hadoop/hadoop.sh

```diff
@@ -213,89 +123,93 @@ fi
         eof
     }
 EOF
-  yarn_jps=$(jps)
-  yarn_suc=$(echo $yarn_jps | grep -w 'NodeManager')
-  yarn_suc1=$(echo $yarn_jps | grep -w 'ResourceManager')
-  if [ "$yarn_suc"x != ""x -a "$yarn_suc1"x != ""x ]; then
-      echo 'yarn run successfully'
-  else 
-      echo 'yarn set up failed'
-      exit 1
-  fi
+
+yarn_jps=$(jps)
+yarn_suc=$(echo $yarn_jps | grep -w 'NodeManager')
+yarn_suc1=$(echo $yarn_jps | grep -w 'ResourceManager')
+if [ "$yarn_suc"x != ""x -a "$yarn_suc1"x != ""x ]; then
+    echo 'yarn run successfully'
+else 
+    echo 'yarn set up failed'
+    exit 1
+fi
+
 popd
-if [ ! $Osname_CentOS -eq 0 ]
-then
-    SetKeyBlankValue()
-    {
-        local sKey1=${1}
-        local sVal1=${2}
-    	local varTxt=${3}
-        local varNDo=${4}
-
-	sKey1=$(sed "s@\([./]\)@\\\\\1@g" <<< "${sKey1}")
-        sVal1=$(sed "s@\([./]\)@\\\\\1@g" <<< "${sVal1}")
-
-        eval local sTxt1=\${${varTxt}}
-
-        grep -q "^[         ]*${sKey1}[     ]\+" <<< "${sTxt1}"
-        if [ $? -ne 0 ]; then
-            printf "%s[%3d]%s[%3d]%5s: [${sKey1}] not found\n" "${FUNCNAME[1]}" "${BASH_LINENO[0]}" "${FUNCNAME[0]}" ${LINENO} "Error"
-            return 1
-        fi
 
-        #f1="/root/caliper/binary/arm_64/hadoop/hibench/conf/99-user_defined_properties.conf"
-        #grep "hibench.hadoop.home          /root/caliper/binary/arm_64/hadoop/hadoop" "${f1}"
-        grep -q "^[         ]*${sKey1}[     ]\+${sVal1}" <<< "${sTxt1}"
-        if [ $? -ne 0 ]; then
-            sTxt1=$(sed "/^[ \t]*${sKey1}[ \t]\+/{s/^\([ \t]*${sKey1}[ \t]\+\).*\$/\1${sVal1}/}" <<< "${sTxt1}")
-            eval ${varTxt}=\${sTxt1}
-            eval let ${varNDo}+=1
-        fi
+##############################
+SetKeyBlankValue()
+{
+    local sKey1=${1}
+    local sVal1=${2}
+    local varTxt=${3}
+    local varNDo=${4}
 
-        return 0
-    }
+    sKey1=$(sed "s@\([./]\)@\\\\\1@g" <<< "${sKey1}")
+    sVal1=$(sed "s@\([./]\)@\\\\\1@g" <<< "${sVal1}")
+
+    eval local sTxt1=\${${varTxt}}
+
+    grep -q "^[ 	]*${sKey1}[ 	]\+" <<< "${sTxt1}"
+    if [ $? -ne 0 ]; then
+        printf "%s[%3d]%s[%3d]%5s: [${sKey1}] not found\n" "${FUNCNAME[1]}" "${BASH_LINENO[0]}" "${FUNCNAME[0]}" ${LINENO} "Error"
+        return 1
+    fi
 
+    #f1="/root/caliper/binary/arm_64/hadoop/hibench/conf/99-user_defined_properties.conf"
+    #grep "hibench.hadoop.home		/root/caliper/binary/arm_64/hadoop/hadoop" "${f1}"
+    grep -q "^[ 	]*${sKey1}[ 	]\+${sVal1}" <<< "${sTxt1}"
+    if [ $? -ne 0 ]; then
+        sTxt1=$(sed "/^[ \t]*${sKey1}[ \t]\+/{s/^\([ \t]*${sKey1}[ \t]\+\).*\$/\1${sVal1}/}" <<< "${sTxt1}")
+        eval ${varTxt}=\${sTxt1}
+        eval let ${varNDo}+=1
+    fi
 
-    AbleKey()
-    {
-        local sKey1=${1}
-        local bAble=${2}
-        local varTxt=${3}
-        local varNDo=${4}
+    return 0
+}
 
-        sKey1=$(sed "s@\([./]\)@\\\\\1@g" <<< "${sKey1}")
+AbleKey()
```

Comment:
it need deal char '/' and '.', and we need judage the key exist first.
if not the key, need remind us, and if the key's value is the same,
we not need modify it again.